pkg/runtime/processor/components.go

```diff
@@ -44,11 +77,45 @@ func (p *Processor) Init(ctx context.Context, comp componentsapi.Component) erro
 		return errors.Join(err, p.compStore.DropPendingComponent())
 	}
 
-	return p.compStore.CommitPendingComponent()
+	if err := p.compStore.CommitPendingComponent(); err != nil {
+		return fmt.Errorf("error committing component: %w", err)
+	}
+
+	return nil
 }
 
-// Close closes the component.
+// Close closes the component and reports the result.
 func (p *Processor) Close(comp componentsapi.Component) error {
+	closeErr := p.internalClose(comp)
+
+	// after performing the initialization, report the result
+	condition := operatorv1.ResourceConditionStatus_SUCCESS
+	var reason, message *string
+	if closeErr != nil {
+		condition = operatorv1.ResourceConditionStatus_FAILURE
+		reason = ptr.Of("ERROR")
+		message = ptr.Of(closeErr.Error())
+	}
+
+	if err := p.reporter(context.TODO(), comp,
```

Comment:
How much work is it to update `Close` to accept a context? Ideally we should.

Reply:
checking

Reply:
seemed easy and added it, tagged you on a bit that i'm not totally sure